geopy/geocoders/geonames.py

```diff
@@ -171,13 +189,67 @@ def reverse(
             'lng': lng,
             'username': self.username
         }
-        url = "?".join((self.api_reverse, urlencode(params)))
+        url = "?".join((self.nearby_api, urlencode(params))) if find_nearby \
+            else "?".join((self.api_reverse, urlencode(params)))
         logger.debug("%s.reverse: %s", self.__class__.__name__, url)
         return self._parse_json(
             self._call_geocoder(url, timeout=timeout),
             exactly_one
         )
 
+    def timezone(
+        self,
+        location,
+        timeout=DEFAULT_SENTINEL
+    ):
+        """
+        Finds the timezone a `location` was in
+        and returns a pytz timezone object.
+
+        :param location: The coordinates for which you want a timezone.
+        :type location: :class:`geopy.point.Point`, list or tuple of (latitude,
+            longitude), or string as "%(latitude)s, %(longitude)s"
+
+        :param int timeout: Time, in seconds, to wait for the geocoding service
+            to respond before raising a :class:`geopy.exc.GeocoderTimedOut`
+            exception. Set this only if you wish to override, on this call
+            only, the value set during the geocoder's initialization.
+
+        :rtype: pytz timezone. See :func:`pytz.timezone`.
+        """
+        if not pytz_available:
+            raise ImportError(
+                'pytz must be installed in order to locate timezones. '
+                ' Install with `pip install geopy -e ".[timezone]"`.'
+            )
+        try:
+            lat, lng = self._coerce_point_to_string(location).split(',')
+        except ValueError:
+            raise ValueError("Must be a coordinate pair or Point")
+
+        params = {
+            "lat": lat,
+            "lng": lng,
+            "username": self.username,
+        }
+
+        url = "?".join((self.tz_api, urlencode(params)))
+
+        response = self._call_geocoder(url, timeout=timeout)
+        try:
+            tz = timezone(response["timezoneId"])
+        except UnknownTimeZoneError:
+            raise GeocoderParseError(
+                "pytz could not parse the timezone identifier (%s) "
+                "returned by the service." % response["timeZoneId"]
+            )
+        except KeyError:
```

Comment:
Too broad code sample is covered by this except clause. I.e. a KeyError raised by the `timezone` function is rather unexpected and shouldn't be processed by this clause.